test/hotspot/jtreg/vmTestbase/nsk/share/jpda/DebugeeProcess.java

```diff
@@ -196,18 +196,27 @@ public void sendSignal(String signal) {
      */
     public int waitFor () {
         long timeout = binder.getArgumentHandler().getWaitTime() * 60 * 1000;
-        int exitCode = 0;
         try {
-            exitCode = waitForDebugee();
+            int exitCode = waitForDebugee();
+            return exitCode;
```

Comment:
I don't think I've ever run across a try block with a return statement before, especially when there is also a finally block. The reader is likely to miss the fact that before the return is done the finally block is executed. It's also odd because now there is no return statement at the end of the method. Although the compiler is smart enough to recognize that this is ok, it is another point of confusion for the reader. Any reason not to just instead do the return at the end of the method?

Reply:
Yes, the code become too unreadable. Moved return out of try/catch.